octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/util/OpenDetailTabEntityMouseListener.java

```diff
@@ -1,72 +1,107 @@
-            }
-            
-            if(Entity.FEATURE == Entity.getEntityType(entityModel) || Entity.EPIC == Entity.getEntityType(entityModel)) {
-            	entityService.openInBrowser(entityModel);
-            } else {
-	            Long id = Long.parseLong(entityModel.getValue("id").getValue().toString());
-	            EntityModelEditorInput entityModelEditorInput = new EntityModelEditorInput(id, Entity.getEntityType(entityModel));
-	            try {
-	                logger.log(new Status(Status.INFO, Activator.PLUGIN_ID, Status.OK, entityModelEditorInput.toString(), null));                
-	                page.openEditor(entityModelEditorInput, EntityModelEditor.ID);                                      
-	            } catch (PartInitException ex) {
-	                logger.log(
-	                        new Status(Status.ERROR, Activator.PLUGIN_ID, Status.ERROR, "An exception has occured when opening the editor", ex));
-	            }
-            }
-            
-        }
-    }
-
-}
+/*******************************************************************************
+ * © 2017 EntIT Software LLC, a Micro Focus company, L.P.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+package com.hpe.octane.ideplugins.eclipse.ui.util;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.core.runtime.ILog;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.swt.events.MouseEvent;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+
+import com.hpe.adm.nga.sdk.model.EntityModel;
+import com.hpe.adm.octane.ideplugins.services.EntityService;
+import com.hpe.adm.octane.ideplugins.services.filtering.Entity;
+import com.hpe.adm.octane.ideplugins.services.mywork.MyWorkUtil;
+import com.hpe.adm.octane.ideplugins.services.util.Util;
+import com.hpe.octane.ideplugins.eclipse.Activator;
+import com.hpe.octane.ideplugins.eclipse.ui.editor.EntityModelEditor;
+import com.hpe.octane.ideplugins.eclipse.ui.editor.EntityModelEditorInput;
+import com.hpe.octane.ideplugins.eclipse.ui.entitylist.EntityMouseListener;
+
+public class OpenDetailTabEntityMouseListener implements EntityMouseListener {
+	
+	private static final Set<Entity> whitelistParentDetails = new HashSet<Entity>();
```

Comment:
@vladbenchea Why do we need 3 places to store this information? Perhaps we can use a common location to store only one instance of this set

Reply:
Perhaps a better name for 'whitelistParentDetails' would be 'supported entities that allow a detail view'

Reply:
Shouldn't the requirement entity also be a part of this set?

Reply:
@paul-ciobanu I added the requirement to be a part of the set. Also I put the set in EntityFieldsConstants, so that we don't have to declare it in 3 different places. It got renamed to the name you suggested "supportedEntitiesThatAllowDetailView".